src/modules/Game.class.js

```diff
@@ -6,63 +6,261 @@
+            pushed = true;
+            i = j;
+            break;
+          }
+        }
+
+        if (!pushed) {
+          newRow.push(current);
+          i++;
+        }
+      } else {
+        i++;
+      }
+    }
+
+    while (newRow.length < row.length) {
+      newRow.push(0);
+    }
+
+    return newRow;
+  }
+
+  rotateMatrixClockwise(matrix) {
+    const n = matrix.length;
+    const rotatedMatrix = [];
+
+    for (let i = 0; i < n; i++) {
+      rotatedMatrix.push([]);
+
+      for (let j = 0; j < n; j++) {
+        rotatedMatrix[i].unshift(matrix[j][i]);
+      }
+    }
+
+    return rotatedMatrix;
+  }
+
+  rotateMatrixCounteClockwise(matrix) {
+    const n = matrix.length;
+    const rotatedMatrix = [];
+
+    for (let i = 0; i < n; i++) {
+      rotatedMatrix.unshift([]);
+
+      for (let j = 0; j < n; j++) {
+        rotatedMatrix[0].push(matrix[j][i]);
+      }
+    }
+
+    return rotatedMatrix;
+  }
+
+  getScore() {
+    return this.score;
+  }
+
+  getState() {
+    return this.state;
+  }
+
+  getStatus() {
+    return this.status;
+  }
+
+  start() {
+    this.status = Game.gameStatuses.playing;
+    this.addCells(2);
+  }
+
+  restart() {
+    this.status = Game.gameStatuses.idle;
+    this.resetState();
+  }
+
+  getEmptyCells() {
+    return this.state
+      .flatMap((row, rowIndex) => {
+        return row.map((cell, colIndex) => {
+          return cell === 0 ? [rowIndex, colIndex] : null;
+        });
+      })
+      .filter((cell) => cell !== null);
+  }
+
+  createNewTile() {
+    const emptyCells = this.getEmptyCells();
+
+    if (!emptyCells.length) {
+      return;
+    }
+
+    const [row, col] =
+      emptyCells[Math.trunc(Math.random() * emptyCells.length)];
+
+    this.state[row][col] = Math.random() < 0.9 ? 2 : 4;
+  }
+
+  addCells(count = 1) {
+    for (let i = 0; i < count; i++) {
+      this.createNewTile();
+    }
+
+    const state = this.getState();
+
+    if (this.isVictory(state)) {
+      this.status = Game.gameStatuses.win;
+    } else if (this.isGameOver(state)) {
+      this.status = Game.gameStatuses.lose;
+    }
+  }
+
+  resetState() {
+    this.state = this.initialState.map((row) => [...row]);
+    this.score = 0;
+  }
+
+  isGameOver(currentState) {
+    if (this.status !== Game.gameStatuses.playing) {
+      return false;
+    }
+
+    for (let row = 0; row < currentState.length; row++) {
+      for (let col = 0; col < currentState[row].length; col++) {
+        if (
+          currentState[row][col] === 0 ||
+          (col < currentState[row].length - 1 &&
+            currentState[row][col] === currentState[row][col + 1]) ||
+          (row < currentState.length - 1 &&
+            currentState[row][col] === currentState[row + 1][col])
+        ) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  }
+
+  updateGameState(state) {
+    this.state = state;
+  }
+
+  updateGameScore(scoreToAdd) {
+    this.score += scoreToAdd;
+  }
+
+  isVictory(state) {
```

Comment:
The 'isVictory' function checks for a win condition, which is having a tile with the value 2048. Ensure that this check is performed correctly and at appropriate times in the game logic.